tests/system/feature_store/test_feature_store.py

```diff
@@ -2575,6 +2575,75 @@ def test_get_offline_for_two_feature_set_with_same_column_name(self):
         for key in res.to_dataframe().to_dict().keys():
             assert key in expected
 
+    @pytest.mark.parametrize("engine", ["local", "dask"])
+    def test_get_offline_features_with_filter(self, engine):
+        engine_args = {}
+        if engine == "dask":
+            dask_cluster = mlrun.new_function(
+                "dask_tests", kind="dask", image="mlrun/ml-models"
+            )
+            dask_cluster.apply(mlrun.mount_v3io())
+            dask_cluster.spec.remote = True
+            dask_cluster.with_requests(mem="2G")
+            dask_cluster.save()
+            engine_args = {
+                "dask_client": dask_cluster,
+                "dask_cluster_uri": dask_cluster.uri,
+            }
+
+        data = pd.DataFrame(
+            {
+                "name": ["A", "B", "C", "D", "E"],
+                "age": [33, 4, 76, 90, 24],
+                "department": ["IT", "RD", "RD", "Marketing", "IT"],
+            },
+            index=[0, 1, 2, 3, 4],
+        )
+        data["id"] = data.index
+
+        one_hot_encoder_mapping = {
+            "department": list(data["department"].unique()),
+        }
+        data_set = FeatureSet(
+            "fs-new", entities=[Entity("id")], description="feature set"
+        )
+        data_set.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+        data_set.set_targets()
+        data_set.plot(rankdir="LR", with_targets=True)
+        fs.ingest(data_set, data, infer_options=fs.InferOptions.default())
+
+        fv_name = "new-fv"
+        features = [
+            "fs-new.name",
+            "fs-new.age",
+            "fs-new.department_RD",
+            "fs-new.department_IT",
+            "fs-new.department_Marketing",
+        ]
+
+        my_fv = fs.FeatureVector(fv_name, features, description="my feature vector")
+        my_fv.save()
+        train_dataset = fs.get_offline_features(
+            fv_name,
+            target=ParquetTarget(),
+            query="age>6 and department_RD==1",
```

Comment:
Please add tests that cover more advanced filter usages and may be problematic in terms of escaping characters - these will be useful for reference as well later on. For example using strings and `in` filters, such as:
`query="name='B' and age<50"`
or
`query="name in ['A','C']"` (not sure this is even how it's supposed to be written, that's why I want this test...)